File: src/types/UploadStatus.ts
/**
 * Upload status of a recording in the database.
 *
 * Maps to Rust enum: src-tauri/src/models/upload_status.rs
 */
export enum UploadStatus {
  Unknown = "unknown",
  Pending = "pending",
  Uploaded = "uploaded",
  Deleted = "deleted",
}

File: src-tauri/src/models/upload_status.rs
use serde::{Deserialize, Serialize};

/// Upload status of a recording in the database.
///
/// Ported from Recorder.Core/Models/UploadStatus.cs
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum UploadStatus {
    Unknown,
    Pending,
    Uploaded,
    Deleted,
}
